### DISCUSSION TITLE
mitmweb can't recognize my upstream proxy formate

### DISCUSSION THREAD

**Discussion by guyujia** [mitmproxy/mitmproxy#7233] [Q&A]
#### Problem Description
I code a demo by mitmproxy, the .py code like
```
from mitmproxy import http
def response(flow: http.HTTPFlow):
    print(f"Intercepted response from {flow.request.url}")
    if 'google' in flow.request.url:
        print(f"Intercepted response from {flow.request.url}")
```
And I used command like 'mitmweb -q -s demo.py --mode upstream://user-10019679_27-country-KR-session-551067:password@proxy-as.ipoasis.com:8888', but it always return error 'Invalid server specification: //user-10019679_27-country-KR-session-551067:password@proxy-as.ipoasis.com:8888', is there any mistake in my script or command?

#### System Information
Mitmproxy: 5.3.0
Python:    3.7.0
OpenSSL:   OpenSSL 1.1.1h  22 Sep 2020
Platform:  Windows-10-10.0.19041-SP0

**Comment by mhils** [comment]
https://docs.mitmproxy.org/stable/concepts-modes/#upstream-proxy

You want `upstream:http://hostname` together with https://docs.mitmproxy.org/stable/concepts-options/#upstream_auth for the authentication.

**Reply by guyujia** [reply]
I figure it out，thx

**Comment by guyujia** [comment]
I got it sir，could u help me to write one example？Cause all information online let me to set the config.yaml up with the formate like
```

# ~/.mitmproxy/config.yaml

# Set up upstream proxy
upstream: "http://user:password@proxy.example.com:8080"

# Optionally, set other configurations like:
listen_port: 8081  # Custom port for mitmweb if needed
set:
  ssl_insecure: true  # To ignore SSL verification (only if necessary)

```
